### ISSUE TITLE
compile_nginx_source resource is alwayws executed

### ISSUE THREAD

**Issue by axl89** [sous-chefs/nginx#458]
### Cookbook version
8.1.2

### Chef-client version
14.1.12

### Platform Details
Kitchen v1.20.0 virtualizing an Ubuntu 16.04 from Bento

### Scenario:
`nginx::source` recipe executes the `compile_nginx_source` every time.

### Steps to Reproduce:
Add the following attributes:

```
default['nginx']['install_method'] = 'source'
default['nginx']['devel']['version'] = '0.3.0'
default['nginx']['devel']['url'] = 'https://github.com/simpl/ngx_devel_kit/archive/v0.3.0.tar.gz'
default['nginx']['devel']['checksum'] = '88e05a99a8a7419066f5ae75966fb1efc409bad4522d14986da074554ae61619'

default['nginx']['lua']['version'] = '0.10.13'
default['nginx']['lua']['url'] = 'https://github.com/chaoslawful/lua-nginx-module/archive/v0.10.13.tar.gz'
default['nginx']['lua']['checksum'] = 'ecea8c3d7f69dd48c6132498ddefb5d83ba9f387fa3d4da14e2abeacdfc8a3ee'

default['nginx']['luajit']['version'] = '2.0.5'
default['nginx']['luajit']['url'] = 'http://luajit.org/download/LuaJIT-2.0.5.tar.gz'
default['nginx']['luajit']['checksum'] = '874b1f8297c697821f561f9b73b57ffd419ed8f4278c82e05b48806d30c1e979'

default['nginx']['source']['default_configure_flags'] = %W(--prefix=#{node['nginx']['source']['prefix']} --sbin-path=#{node['nginx']['source']['sbin_path']} --with-ld-opt='-Wl,-rpath,/usr/local/lib' --modules-path=/usr/lib/nginx/modules --conf-path=#{node['nginx']['dir']}/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --http-client-body-temp-path=/var/cache/nginx/client_temp --http-proxy-temp-path=/var/cache/nginx/proxy_temp --http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp --http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp --http-scgi-temp-path=/var/cache/nginx/scgi_temp --user=#{node['nginx']['user']} --group=#{node['nginx']['group']} --with-http_image_filter_module --with-compat --with-file-aio --with-threads --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-mail --with-mail_ssl_module --with-stream --with-stream_realip_module --with-stream_ssl_module --with-stream_ssl_preread_module)

default['nginx']['source']['modules']  = %w(
  nginx::http_ssl_module
  nginx::http_gzip_static_module
  nginx::ngx_devel_module
  nginx::ngx_lua_module
)
```

### Expected Result:
The `compile_nginx_source` resource should not be executed every time. Only the first one.

### Actual Result:
No error, but every converge, the following message is shown:
```
Recipe: nginx::source
         * bash[compile_nginx_source] action run
           - execute "bash"  "/tmp/chef-script20180530-11132-1blbfx1"
```

**Comment by axl89** [comment]
I think the issue lives in [source.rb line 101](https://github.com/chef-cookbooks/nginx/blob/af7b9d32780e367ffef8cb50aceba61e7b9109a5/recipes/source.rb#L101). After checking the module recipes (such as lua or devel), the `configure_arguments` attribute is never updated.

Then, since `configure_arguments` and `default_configure_flags` end up being different, nginx compiles again (the `not_if` condition is not met).

**Comment by damacus** [comment]
This should be fixed via the `v10.0.0` release.

If this isn't the case please reopen this issue with some updated reproducer steps. 

Thanks,
Dan
